zipline/testing/core.py

```diff
@@ -1470,6 +1475,188 @@ def ensure_doctest(f, name=None):
     return f
 
 
+def run_correlations(pearson_factor,
```

Comment:
I'm nervous about two things here:

1. This is a **lot** of logic to be sharing between tests.  Do we really want to be running all of this in two places?  What useful functionality is being by running this twice?

2. There's a lot of logic in here that looks like it's making assumptions about the particular test data that's being passed in here (e.g. indexing with `int(asset) - `).